weave

```diff
@@ -1846,11 +1846,17 @@ EOF
         deprecation_warnings "$@"
         check_not_running router $CONTAINER_NAME        $BASE_IMAGE
         check_not_running proxy  $PROXY_CONTAINER_NAME  $BASE_EXEC_IMAGE
-        check_not_running plugin $PLUGIN_CONTAINER_NAME $BASE_PLUGIN_IMAGE
         COMMON_ARGS=$(common_launch_args "$@")
         launch_router "$@"
         launch_proxy  $COMMON_ARGS
-        plugin_disabled || launch_plugin $COMMON_ARGS
+        # Allow an existing plugin to keep running, because we run it with restart=always
+        if ! plugin_disabled ; then
+            if check_not_running plugin $PLUGIN_CONTAINER_NAME $BASE_PLUGIN_IMAGE ; then
+                launch_plugin $COMMON_ARGS
+            else
+                [ $? -lt 2 ] || exit 1
+            fi
+        fi
```

Comment:
You could turn this into just
```
plugin_disabled || ! check_not_running ... || launch_plugin ...
```
by making `check_not_running` exit (as it used to) in the cases where you currently return 2 or 3.

Whether that is better is debatable.